Source/ARTRealtimeChannel.m

```diff
@@ -603,14 +610,11 @@ - (void)transition:(ARTRealtimeChannelState)state status:(ARTStatus *)status {
         case ARTRealtimeChannelSuspended:
             [_attachedEventEmitter emit:nil with:status.errorInfo];
             if (self.realtime.shouldSendEvents) {
-                channelRetryListener = [self unlessStateChangesBefore:self.realtime.options.channelRetryTimeout do:^{
+                NSTimeInterval retryDelay = [ARTUtils retryDelayFromInitialRetryTimeout:self.realtime.options.channelRetryTimeout
+                                                                         forRetryNumber:++_retryCount];
+                channelRetryListener = [self unlessStateChangesBefore:retryDelay do:^{
                     [self.realtime.logger debug:__FILE__ line:__LINE__ message:@"RT:%p C:%p (%@) reattach initiated by retry timeout", self->_realtime, self, self.name];
-                    [self reattachWithReason:nil callback:^(ARTErrorInfo *errorInfo) {
-                        if (errorInfo) {
-                            ARTStatus *status = [ARTStatus state:ARTStateError info:errorInfo];
```

Comment:
Why have we removed this code that sets the channel to suspended? I haven't looked into its purpose but at a glance it doesn't seem related to the intention of this pull request.

Reply:
Here is an explanation https://github.com/ably/ably-cocoa/pull/1520/commits/21b207f5f2926bf72ef7c961e8a0ea27feb46047

Reply:
Could you open this one as a separate PR? I think that if it's a bug fix then we should give it a proper review separately.

Reply:
https://github.com/ably/ably-cocoa/pull/1551

Reply:
👍 thanks!